### PR TITLE
Chore/upgrade deps (with CRA update)

### PR DESCRIPTION
**Pull request by satazor**
The PR #223 updated all the project dependencies which broke the build process and many other stuff. The reason is that a lot of CRA files needed to be updated as well.

- updated config/ but kept stylus and rollbar configs
- updated scripts/
- updated package.json
- fixed tests
- fixed dynamic imports usage
- fixed error handling in Home component

What needs to be tested:

- [ ] Run the app in development and see if it works correctly *in all compatible browsers*
- [ ] Build the project and run the app in production and see if it works correctly *in all compatible browsers* (you may use a local static server for this)
- [ ] Test a version deployed via the CI to see if everything works correctly
- [ ] Check if Rollbar is still working as expected, including the source-maps

It's important dedicate some time in testing all the updated stuff. More specifically, there were packages that were updated to new major versions.

What needs to be done next (create issues for these):

- [ ] Fix linting errors
- [ ] Integrate standard linting into CRA instead of using eslint
- [ ] `npm run test:e2e` does not run anything; this is happening in master

@pgte feels free to close this and cherry-pick the commit to #223.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,11 +1,12 @@
 {
   "name": "peerpad",
   "version": "0.3.1",
+  "homepage": "./",
   "private": true,
   "leadMaintainer": "Jim Pick <jim@jimpick.com>",
   "scripts": {
     "start": "node scripts/start.js",
-    "build": "GENERATE_SOURCEMAP=true GIT_COMMIT=$(git rev-parse HEAD) PUBLIC_URL='./' node scripts/build.js",
+    "build": "GIT_COMMIT=$(git rev-parse HEAD) node scripts/build.js",
```
Comment (pgte) [inline package.json]: is removing the env vars intentional?

Comment (satazor) [reply package.json]: Yes, they are:

- `GENERATE_SOURCEMAP` is used only once in `webpack.config.prod.js`:

```js
const shouldUseSourceMap = process.env.GENERATE_SOURCEMAP !== 'false';
``` 

It's default value is `true`, so there's no need to specify it there.

- `PUBLIC_URL` was removed in favor of the `homepage` property in the package.json as it is the recommend way to do it.

Comment (pgte) [reply package.json]: @satazor gotcha, thank you!

**Comment by pgte** [comment]
@satazor Awesome, thank, you, merged into #223. Closing.

**Pull request closed without merging**